pdfview.js

```diff
@@ -1,12 +1,16 @@
-// Get PDF link
-inside_span = document.getElementsByClassName("attachments-file-name")[0].innerHTML
+try{
+    // Get PDF link
+    inside_span = document.getElementsByClassName("attachments-file-name")[0].innerHTML
 
-start = inside_span.indexOf("<a href=\"") + 9
-end = inside_span.indexOf(">") - 28
-link = inside_span.slice(start, end)
+    start = inside_span.indexOf("<a href=\"") + 9
+    end = inside_span.indexOf(">") - 28
+    link = inside_span.slice(start, end)
 
-// Change iframe
-document.getElementsByClassName("docviewer-iframe")[0].setAttribute("src", link)
+    if (link.slice(-4)!=".pdf"){throw Error}
```

Comment:
Is there a better way to check the type of a file than it's extension.

Reply:
this is a really good way, prob the best way since the main way to get what type of file it is is via the file name.

Reply:
And if there is no extension? Is there a way we can check the type of a file?
I understand that this might not be possible, but was wondering if it was.

Reply:
wdym if there is no extension? anothe way would be to look into the request and see what type of application it is

Reply:
> wdym if there is no extension?

For example, if it is a PDF, but the `.pdf` extension is not on the file.
> anothe way would be to look into the request and see what type of application it is

I think this is a great idea, but doesn't schoology's files not have mimetypes.

Reply:
I don't think they have mimetypes. I am almost certain that every pdf that needs to be downloaded will have pdf in its name.

Reply:
Hmmm... I'll confirm that they don't have mimetypes. Until then, this sounds good.